File: try/app/(modules)/crm/dto/order.dto.ts
export interface OrderItem {
  id?: string;
  productId?: string;
  product?: any;
  productName?: string;
  quantity: number;
  unitPrice: number;
  totalPrice?: number;
  discount?: number;
  orderItemSizes?: Array<{ size: string; quantity: number }>;
  orderItemMaterials?: Array<{ materialId: string; requiredQuantity: number; unit: string }>;
}

export interface Order {
  id?: string;
  customerId: string;
  orderNumber: string;
  status: string;
  totalAmount: number;
  dueDate: string;
  orderItems?: OrderItem[];
  items?: OrderItem[];
  createdAt?: string;
  updatedAt?: string;
}

File: backend/Dto/Order/OrderDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Model;
using backend.Dto.Customer;
using backend.Dto.OrderItem;
using backend.Dto.Product;
using backend.Dto.Fabric;
using backend.Dto.WorkCenter;
using backend.Dto.ProductionPlan;
using backend.Dto.ProductionPlanProduct;
using backend.Dto.ProductionPlanProductSize;
using backend.Dto.ProductionPlanStage;
using backend.Dto.Material;
using backend.Dto.BillOfMaterial;
using backend.Dto.Warehouse;
using backend.Dto.WarehouseRoom;
using backend.Dto.WarehouseShelf;
using backend.Dto.Inventory;
using backend.Dto.Outlet;
using backend.Dto.OutletDemand;
using backend.Dto.Transaction;
using backend.Dto.ActivityLog;
using backend.Model.Enums;

namespace backend.Dto.Order
{
    public class OrderDto
    {
        // <crudgen:properties>
        public Guid Id { get; set; }

        public string OrderNumber { get; set; } = string.Empty;

        public OrderStatus Status { get; set; }

        public decimal TotalAmount { get; set; }

        public DateTime DueDate { get; set; }

        [Required(ErrorMessage = "CreatedAt is required.")]
        public DateTime CreatedAt { get; set; }

        public string CreatedBy { get; set; } = string.Empty;

        public DateTime UpdatedAt { get; set; }

        public string UpdatedBy { get; set; } = string.Empty;

        public Guid CustomerId { get; set; }
        [NotMapped]
        public List<OrderItemDto> OrderItems { get; set; } = new List<OrderItemDto>();

        // </crudgen:properties>
    }
}


